### PR TITLE
Qiime

### PR DESCRIPTION
**Pull request by lleroi**
454 qiime workflow xml wrappers .
I deleted the tgz in/output option and choose which options can be specified on the xml wrappers.

For the pick_otus.py wrapper, I modified how the clustering options are displayed (eg. swarm options are displayed if swarm is selected for the clustering, uclust options displayed if uclust is selected etc...)
I selected 6 of the 12 clustering methods (those are the most interesting for our team) : uclust, uclust_ref, swarm, usearch, usearch_ref and sumaclust.

### CODE REVIEW

**Comment by bgruening** [comment]
@lleroi this is unfortunately not mergeable. Please let me know if you need help to resolve this.
You would need to do a "git rebase". Thanks!

**Comment by lleroi** [comment]
@bgruening sorry for the delay... 

now it's mergeable and pass the auto tests !

Hope it will help

**Comment by bgruening** [comment]
Merge early and often :)
@lparsons @pjbriggs I will merge this now and make a review on the main branch so others can help in fixing things. Please rebase.

Thanks a lot @lleroi !!!

File: tools/qiime/align_seqs.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" ?>
+<tool id="1-9_align_seqs" name="10 - align seqs" version="1.9.0">
+	<description>Align sequences using a variety of alignment methods</description>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: @lleroi please fix indentation. We use 4 spaces as indentation. Much like Python pep8.

File: tools/qiime/align_seqs.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	<requirements>
+		<requirement type="package">qiime</requirement>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: This should probably be part of a macro file and should contain a version number.

File: tools/qiime/align_seqs.xml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+#if $min_percent_id:
+ -p $min_percent_id
+#end if
+
+ -o align_seqs_output;
+ 
+</command>
+	<inputs>
+		<param label="-i/--input_fasta_fp: path to the input fasta file" name="input_fasta_fp" optional="False" type="data"/>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: The original parameter should be part of the help="(-i/--input_fasta_fp)" and not part of the label. This looks ugly in the user interface :)

File: tools/qiime/align_seqs.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		<param label="-m/--alignment_method: Method for aligning sequences. Valid choices are: pynast, infernal, clustalw, muscle, infernal, mafft [default: pynast]" name="alignment_method" optional="True" type="select">
+			<option selected="True" value="None">Selection is Optional</option>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: I don't think you need to have this line included. optional="True" will take care of it.

File: tools/qiime/align_seqs.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+			<option value="pynast">pynast</option>
+			<option value="infernal">infernal</option>
+			<option value="clustalw">clustalw</option>
+			<option value="muscle">muscle</option>
+			<option value="infernal">infernal</option>
+			<option value="mafft">mafft</option>
+		</param>
+		<param label="-a/--pairwise_alignment_method: method for performing pairwise alignment in PyNAST. Valid choices are muscle, pair_hmm, clustal, blast, uclust, mafft [default: uclust]" name="pairwise_alignment_method" optional="True" type="select">
+			<option selected="True" value="None">Selection is Optional</option>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: optional="True"

File: tools/qiime/align_seqs.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+			<option value="muscle">muscle</option>
+			<option value="pair_hmm">pair_hmm</option>
+			<option value="clustal">clustal</option>
+			<option value="blast">blast</option>
+			<option value="uclust">uclust</option>
+			<option value="mafft">mafft</option>
+		</param>
+		<param default="/home12/caparmor/bioinfo/softs/sources/Qiime/data/core_set_aligned.fasta.imputed" label="-t/--template_fp: Filepath for template alignment [default: /home12/caparmor/bioinfo/softs/sources/Qiime/data/core_set_aligned.fasta.imputed]" name="template_fp" optional="True" type="data"/>
+		<param default="-1" label="-e/--min_length: Minimum sequence length to include in alignment [default: 75% of the median input sequence length]" name="min_length" optional="True" type="integer"/>
+		<param default="0.75" label="-p/--min_percent_id: Minimum percent sequence identity to closest blast hit to include sequence in alignment [default: 0.75]" name="min_percent_id" optional="True" type="float"/>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: default?
What do you want to do? You can set a default value with `value`

File: tools/qiime/align_seqs.xml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+		<param label="-a/--pairwise_alignment_method: method for performing pairwise alignment in PyNAST. Valid choices are muscle, pair_hmm, clustal, blast, uclust, mafft [default: uclust]" name="pairwise_alignment_method" optional="True" type="select">
+			<option selected="True" value="None">Selection is Optional</option>
+			<option value="muscle">muscle</option>
+			<option value="pair_hmm">pair_hmm</option>
+			<option value="clustal">clustal</option>
+			<option value="blast">blast</option>
+			<option value="uclust">uclust</option>
+			<option value="mafft">mafft</option>
+		</param>
+		<param default="/home12/caparmor/bioinfo/softs/sources/Qiime/data/core_set_aligned.fasta.imputed" label="-t/--template_fp: Filepath for template alignment [default: /home12/caparmor/bioinfo/softs/sources/Qiime/data/core_set_aligned.fasta.imputed]" name="template_fp" optional="True" type="data"/>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: ??

File: tools/qiime/align_seqs.xml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+		<param default="-1" label="-e/--min_length: Minimum sequence length to include in alignment [default: 75% of the median input sequence length]" name="min_length" optional="True" type="integer"/>
+		<param default="0.75" label="-p/--min_percent_id: Minimum percent sequence identity to closest blast hit to include sequence in alignment [default: 0.75]" name="min_percent_id" optional="True" type="float"/>
+	</inputs>
+	<outputs>
+		<data format="txt" from_work_dir="align_seqs_output/*_aligned.fasta" name="output_dir_aligned" label="${alignment_method}_aligned.fasta" />
+        <data format="txt" from_work_dir="align_seqs_output/*_failures.fasta" name="output_dir_failures" label="${alignment_method}_failures.fasta"/>
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: Indentation ...

File: tools/qiime/align_seqs.xml [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        <data format="txt" from_work_dir="align_seqs_output/*_log.txt" name="output_dir_log"  label="s${alignment_method}_log.txt"/>
+	</outputs>
+	<help>
+This script aligns the sequences in a FASTA file to each other or to a template sequence alignment, depending on the method chosen. Currently, there are three methods which can be used by the user:
+
+1. PyNAST (Caporaso et al., 2009) - The default alignment method is PyNAST, a python implementation of the NAST alignment algorithm.  The NAST algorithm aligns each provided sequence (the &quot;candidate&quot; sequence) to the best-matching sequence in a pre-aligned database of sequences (the &quot;template&quot; sequence).  Candidate sequences are not permitted to introduce new gap characters into the template database, so the algorithm introduces local mis-alignments to preserve the existing template sequence.
+
+2. MUSCLE (Edgar, 2004) - MUSCLE is an alignment method which stands for MUltiple Sequence Comparison by Log-Expectation.
+
+3. INFERNAL (Nawrocki, Kolbe, &amp; Eddy, 2009) - Infernal (&quot;INFERence of RNA ALignment&quot;) is for an alignment method for using RNA structure and sequence similarities.
```
Comment (bgruening) [inline tools/qiime/align_seqs.xml]: Please use the <citations> section to reference papers.
Also keep in mind to use CDATA https://github.com/galaxyproject/tools-iuc/blob/99380e401b08df7a7394ba03cc9bed353667300b/tools/collection_column_join/collection_column_join.xml#L75 to not need to HTML code like `&quot;`

File: tools/qiime/alpha_diversity.xml [hunk 10]
```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" ?>
+<tool id="1-9_alpha_diversity" name="alpha diversity" version="1.9.0">
+	<description>Calculate alpha diversity on each sample in an otu table, using a variety of alpha diversity metrics</description>
+	<requirements>
+		<requirement type="package">qiime</requirement>
```
Comment (bgruening) [inline tools/qiime/alpha_diversity.xml]: Macros!

File: tools/qiime/alpha_rarefaction.xml [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+ -e $max_rare_depth
+#end if
+;
+mv $output_dir.files_path/alpha_rarefaction_plots/* $output_dir.files_path/
+</command>
+	<inputs>
+		<param label="-i/--otu_table_fp: the input otu table [REQUIRED]" name="otu_table_fp" optional="False" type="data"/>
+		<param label="-m/--mapping_fp: path to the mapping file [REQUIRED]" name="mapping_fp" optional="False" type="data"/>
+		<param label="-p/--parameter_fp: path to the parameter file, which specifies changes to the default behavior. See http://www.qiime.org/documentation/file_formats.html#qiime-parameters . [if omitted, default values will be used]" name="parameter_fp" optional="True" type="data"/>
+		<param default="10" label="-n/--num_steps: number of steps (or rarefied OTU table sizes) to make between min and max counts [default: 10]" name="num_steps" optional="True" type="integer"/>
```
Comment (bgruening) [inline tools/qiime/alpha_rarefaction.xml]: default should be replaced by value

**Comment by bgruening** [comment]
@lleroi can you address these issue please? I will do a more careful review later.

**Comment by lleroi** [comment]
@bgruening 

I change the indentations and the <requirement>qiime>/requirement> for all the wrappers. (most of the issues are due to the generator script :)) I can't see the modifications here... Do I need to PR again ?

**Comment by nsoranzo** [comment]
@lleroi Yes, this PR has already been merged.

**Comment by lleroi** [comment]
@nsoranzo  thank you !

**Pull request merged**